browser/resources/settings/brave_leo_assistant_page/memory_section.ts

```diff
@@ -0,0 +1,230 @@
+      },
+      showDeleteAllDialog_: {
+        type: Boolean,
+        value: false
+      }
+    }
+  }
+
+  static get observers() {
+    return [
+      'loadMemories_(prefs.brave.ai_chat.user_memories.value)',
+    ]
+  }
+
+  browserProxy_: BraveLeoAssistantBrowserProxy =
+    BraveLeoAssistantBrowserProxyImpl.getInstance()
+  declare memoriesList_: string[]
+  declare editingMemoryItem_: string | null
+  declare showMemoryDialog_: boolean
+  declare editingMemory_: string
+  declare isMemoryInputValid_: boolean
+  declare isMemoryInputError_: boolean
+  declare showDeleteDialog_: boolean
+  declare deleteMemoryItem_: string | null
+  declare showDeleteAllDialog_: boolean
+
+  override ready() {
+    super.ready()
+    this.loadMemories_();
+  }
+
+  loadMemories_() {
+    // Check if prefs and the specific pref are available before accessing
+    if (this.prefs?.brave?.ai_chat?.user_memories?.value) {
+      this.memoriesList_ = [...this.prefs.brave.ai_chat.user_memories.value];
+    } else {
+      // Initialize with empty array if prefs are not available yet
+      this.memoriesList_ = [];
+    }
+  }
+
+  saveMemory_(memoryItem: string, oldItem: string | null) {
+    console.log('saveMemory_ called with oldItem:', oldItem);
+    if (!oldItem) {
+      console.log('Adding new memory');
+      this.appendPrefListItem(USER_MEMORIES_PREF_KEY, memoryItem);
+      // Manually reload memories immediately
+      this.loadMemories_();
+    } else {
+      console.log('Updating memory from:', oldItem, 'to:', memoryItem);
+      this.updatePrefListItem(USER_MEMORIES_PREF_KEY, oldItem, memoryItem);
+      // Manually reload memories immediately
+      this.loadMemories_();
+    }
+  }
+
+  onMemoryDialogSave_(e: { detail: { memoryItem: string } }) {
+    this.showMemoryDialog_ = false
+    this.saveMemory_(e.detail.memoryItem, this.editingMemoryItem_);
+    this.editingMemoryItem_ = null;
+  }
+
+  onMemoryDialogClose_() {
+    this.editingMemoryItem_ = null;
+    this.showMemoryDialog_ = false;
+  }
+
+  handleDelete_(e: any) {
+    this.deleteMemoryItem_ = e.model.item;
+    this.showDeleteDialog_ = true;
+  }
+
+  onDeleteDialogCancel_() {
+    this.showDeleteDialog_ = false;
+    this.deleteMemoryItem_ = null;
+  }
+
+  onDeleteDialogConfirm_() {
+    if (this.deleteMemoryItem_) {
+      this.deletePrefListItem(USER_MEMORIES_PREF_KEY, this.deleteMemoryItem_);
+      this.loadMemories_();
+    }
+    this.showDeleteDialog_ = false;
+    this.deleteMemoryItem_ = null;
+  }
+
+  handleEdit_(e: any) {
+    this.editingMemoryItem_ = e.model.item;
+    this.editingMemory_ = e.model.item || '';
+    this.showMemoryDialog_ = true;
+  }
+
+  handleAddNewMemory_() {
+    this.editingMemoryItem_ = null;
+    this.editingMemory_ = '';
+    this.showMemoryDialog_ = true;
+  }
+
+  onDialogInput_(e: any) {
+    const trimmedValue = e.value.trim()
+    this.editingMemory_ = trimmedValue
+    this.isMemoryInputError_ = trimmedValue.length > 512
```

Comment:
I'm a bit surprised this lets us type spaces. Its being called every keystroke (because we bound to `on-input`)